### PR TITLE
PMD Problems - Utils

### PR DESCRIPTION
**Pull request by Tohaker**
### PMD Problems
`Utils` 
[AvoidFileStream](https://pmd.github.io/pmd-6.17.0/pmd_rules_java_performance.html#avoidfilestream) - line 99
[AvoidReassigningParameters](https://pmd.github.io/pmd-6.17.0/pmd_rules_java_bestpractices.html#avoidreassigningparameters) - line 111
[LocalVariableNamingConventions](https://pmd.github.io/pmd-6.17.0/pmd_rules_java_codestyle.html#localvariablenamingconventions) - line 125

### Solution
Removed the method `copyFromJar` as this was not being used anywhere.
Renamed variables.

### Extras
* Split out date-specific utilities into their own class.
* 90% Test coverage

### CODE REVIEW

**Comment by Tohaker** [comment]
> are you sure that return System.getProperty("os.name").indexOf("indow") > 0; and
> return System.getProperty("os.name").indexOf("win") > 0; give the same answer ? Especially, I am pretty sure that System.getProperty("os.name") return sometime a Win and not a win.

I didn't change this at all, and if the capital is a problem, then the current implementation will work surely?

**Comment by schwabdidier** [comment]
One of my problems is that I was unable to find a list of os.name...

**Comment by Tohaker** [comment]
If it bothers you a lot, we could turn to [Apache SystemUtils](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SystemUtils.html#IS_OS_WINDOWS)?

**Comment by kuhlkrein** [comment]
I used this in the tobiistreamengineforjava and it works fine: 

```
    private static boolean isWindows() {
        return (OS.contains("win"));
    }

    private static boolean isUnix() {
        return (OS.contains("nix") || OS.contains("nux") || OS.contains("aix"));
    }

    private static boolean isMac() {
        return (OS.contains("mac"));
    }
```

File: gazeplay-commons/src/main/java/net/gazeplay/commons/utils/games/Utils.java [hunk 1]
```diff
@@ -17,114 +17,62 @@
...
+        return windowsPath;
     }
 
     /**
-     * @return true if the operating system is a Windows
+     * @return true if the operating system is Windows
      */
     public static boolean isWindows() {
-        return System.getProperty("os.name").indexOf("indow") > 0;
+        return System.getProperty("os.name").indexOf("win") > 0;
```
Comment (kuhlkrein) [inline gazeplay-commons/src/main/java/net/gazeplay/commons/utils/games/Utils.java]: I think you should add the toLowerCase just to be sure
System.getProperty("os.name").toLowerCase().indexOf("win")

**Pull request merged**